### PR TITLE
Bug 1994439: Review page of ODF wizard does not follow console conventions

### PR DESCRIPTION
**Pull request by afreen23**
- Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1994439
 - Fixes selection box in advanced section. There should be a selection and no checkbox.

Before:
Screenshot from 2021-08-17 15-05-19
Screenshot from 2021-08-17 15-06-47

After:
Screenshot from 2021-08-17 15-02-06
Screenshot from 2021-08-17 15-03-41

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/advanced-section.tsx [hunk 1]
```diff
@@ -1,48 +1,61 @@
...
 }) => {
   const { t } = useTranslation();
+  const [isSelectOpen, setIsSelectOpen] = React.useState(false);
 
-  const handleSelection: CheckboxProps['onChange'] = (checked) => {
+  const handleSelection: SelectProps['onSelect'] = (_, value) => {
     dispatch({
       type: 'backingStorage/setDeployment',
-      payload: checked ? DeploymentType.MCG : DeploymentType.ALL,
+      payload: value as DeploymentType,
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/advanced-section.tsx:29]: On `SelectProps['onSelect']` the value is string hence not matching with payload which is of "DeploymentType"

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx [hunk 2]
```diff
@@ -192,7 +192,7 @@ export const BackingStorage: React.FC<BackingStorageProps> = ({
           id={`bs-${BackingStorageType.LOCAL_DEVICES}`}
         />
         <Radio
-          label={t('ceph-storage-plugin~Connect a new external storage system')}
+          label={t('ceph-storage-plugin~Connect an external storage platform')}
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx:195]: UXD change. Everything is termed as "storage platform"

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 3]
```diff
@@ -1,224 +1,128 @@
...
-  BlueInfoCircleIcon,
-  RedExclamationCircleIcon,
-  useFlag,
-} from '@console/shared/src';
+import { TextContent, Text, TextVariants, List, ListItem } from '@patternfly/react-core';
+import { useFlag } from '@console/shared/src';
 import { humanizeBinaryBytes } from '@console/internal/components/utils';
-import { MINIMUM_NODES, NetworkTypeLabels, NO_PROVISIONER } from '../../../../constants';
+import { BackingStorageType } from '../../../../constants/create-storage-system';
+import { NetworkTypeLabels, NO_PROVISIONER } from '../../../../constants';
```
Comment (TimothyAsirJeyasing) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:7]: Any reason why some constants are in Pascal Case? NetworkTypeLabels

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 4]
```diff
@@ -1,224 +1,128 @@
...
+    createLocalVolumeSet,
+    backingStorage,
+  } = state;
+  const { nodes, capacity } = capacityAndNodes;
+  const { encryption, kms, networkType } = securityAndNetwork;
+  const { deployment, externalStorage, type } = backingStorage;
 
-  const validations = [...capacityAndNodesValidate(nodes, enableArbiter)];
   const isNoProvisioner = storageClass.provisioner === NO_PROVISIONER;
+  const formattedCapacity = !isNoProvisioner ? `${OSD_CAPACITY_SIZES[capacity]} TiB` : capacity;
```
Comment (TimothyAsirJeyasing) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:44]: Is there any possibility for the size to goes bellow TiB ?

**Review by TimothyAsirJeyasing — COMMENTED** [review]
LGTM few minor comments

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 3 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:7]: NetworkTypeLabels is an enum. I have not created that

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 4 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:44]: No, we dont support that

**Comment by yuvalgalanti** [comment]
looks great, thanks for fixing that!

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 4 above)
Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:44]: Is `Requested Capacity` != `Cluster Capacity` ??

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:44]: yes it is

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 5]
```diff
@@ -1,224 +1,128 @@
...
-  [AlertVariant.success]: GreenCheckCircleIcon,
-  [AlertVariant.warning]: YellowExclamationTriangleIcon,
-  [AlertVariant.info]: BlueInfoCircleIcon,
-  [AlertVariant.danger]: RedExclamationCircleIcon,
-};
-
-export const ReviewListTitle: React.FC<ReviewListTitleProps> = ({ text }) => (
-  <dt>
+export const ReviewItem = ({ children, title }) => (
+  <div className="odf-review-and-create__review-item">
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:20]: odf-review-and-create__review--item I saw above you used double hyphen symbol :)

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 6]
```diff
@@ -1,224 +1,128 @@
...
+  } = state;
+  const { nodes, capacity } = capacityAndNodes;
+  const { encryption, kms, networkType } = securityAndNetwork;
+  const { deployment, externalStorage, type } = backingStorage;
 
-  const validations = [...capacityAndNodesValidate(nodes, enableArbiter)];
   const isNoProvisioner = storageClass.provisioner === NO_PROVISIONER;
+  const formattedCapacity = !isNoProvisioner ? `${OSD_CAPACITY_SIZES[capacity]} TiB` : capacity;
+
+  const hasEncryption = encryption.clusterWide || encryption.storageClass;
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:46]: this is fine, one more alternative is   encryption.hasHandled

**Comment by TimothyAsirJeyasing** [comment]
LGTM

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 7]
```diff
@@ -1,224 +1,128 @@
...
+        </ListItem>
+        {type === BackingStorageType.EXTERNAL && (
+          <ListItem>
+            {t('ceph-storage-plugin~External storage platform: {{storagePlatform}}', {
+              storagePlatform,
+            })}
+          </ListItem>
+        )}
+      </ReviewItem>
+      <ReviewItem title={t('ceph-storage-plugin~Capacity and nodes')}>
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:79]: can we consider nodes as CRD names here? is it Nodes or nodes?

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 5 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:20]: I cant find that occurrence, can you point line no ?

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 6 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:46]: I checked its for validations and by default always "true". Here we need to know if an encryption of any level is enabled or not.

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 7 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:79]: This is to match with navigation title. We also not capitalizing storage class if you check.

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:79]: I checked that from console and pf teams. Its correct.

**Review by SanjalKatiyar — APPROVED** [review]
/lgtm

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 6 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:46]: acl

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 7 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:79]: ack

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 5 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:20]: https://github.com/openshift/console/pull/9815/files#diff-f3f8fdf9bc309dbd09ce0bcc4b050057d4e80509065e52d4fb110be412821b26R51

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:20]: odf-backing-storage__selection--width

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:20]: so we use it only for a property?

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:20]: I think `review-item` is one word that is why may be ?

**Pull request merged**